Content.Server/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,197 @@
+            seedProto.HighPressureTolerance, seedProto.PestTolerance, seedProto.WeedTolerance, traits, seedProto.MutationPrototypes);
+    }
+
+    /// <summary>
+    ///     Analysis of unique seed.
+    /// </summary>
+    public PlantAnalyzerScannedSeedPlantInformation ObtainingGeneDataSeed(SeedData seed, EntityUid target, bool trayChecker, Boolean scanMode)
+    {
+        string plantHarvestType = "";
+        if (seed.HarvestRepeat == HarvestType.Repeat) plantHarvestType = HarvestType.Repeat.ToString().ToLower();
+        if (seed.HarvestRepeat == HarvestType.NoRepeat) plantHarvestType = HarvestType.NoRepeat.ToString().ToLower();
+        if (seed.HarvestRepeat == HarvestType.SelfHarvest) plantHarvestType = HarvestType.SelfHarvest.ToString().ToLower();
+
+        string exudeGases = new StringBuilder("").AppendJoin("\n   ", seed.ExudeGasses.Select(item => item.Key.ToString())).ToString();
+        string seedChem = new StringBuilder("\n   ").AppendJoin("\n   ", seed.Chemicals.Select(item => item.Key.ToString())).ToString();
+        string speciation = new StringBuilder("").AppendJoin("\n   ", seed.MutationPrototypes.Select(item => item.ToString())).ToString();
+        string traits = new StringBuilder("\n   ").AppendJoin("\n   ", CheckAllMutations(seed).Select(item => item.ToString())).ToString();
+
+        return new PlantAnalyzerScannedSeedPlantInformation(GetNetEntity(target), scanMode, trayChecker,
+            Loc.GetString(seed.DisplayName), seedChem, plantHarvestType, exudeGases, seed.Endurance,
+            seed.Yield, seed.Lifespan, seed.Maturation, seed.GrowthStages, seed.Potency,
+            seed.NutrientConsumption, seed.WaterConsumption, seed.IdealHeat, seed.HeatTolerance,
+            seed.IdealLight, seed.LightTolerance, seed.ToxinsTolerance, seed.LowPressureTolerance,
+            seed.HighPressureTolerance, seed.PestTolerance, seed.WeedTolerance, traits, seed.MutationPrototypes);
+    }
+
+    /// <summary>
+    ///     Returns information about the seeds boolean mutations.
+    /// </summary>
+    public List<string> CheckAllMutations(SeedData plant)
+    {
+        List<string> mutationsList = new List<string>();
+        if (plant.TurnIntoKudzu) mutationsList.Add(Loc.GetString("plant-analyzer-mutation-turnintokudzu"));
+        if (plant.Seedless) mutationsList.Add(Loc.GetString("plant-analyzer-mutation-seedless"));
+        if (plant.Slip) mutationsList.Add(Loc.GetString("plant-analyzer-mutation-slip"));
+        if (plant.Sentient) mutationsList.Add(Loc.GetString("plant-analyzer-mutation-sentient"));
+        if (plant.Ligneous) mutationsList.Add(Loc.GetString("plant-analyzer-mutation-ligneous"));
+        if (plant.Bioluminescent) mutationsList.Add(Loc.GetString("plant-analyzer-mutation-bioluminescent"));
+        if (plant.CanScream) mutationsList.Add(Loc.GetString("plant-analyzer-mutation-canscream"));
```

Comment:
Don't do `Loc.GetString` on server if you can avoid it. Just send the bools and have client do it.

Reply:
I was told not to use "randomly" indexed lists, wouldn't this be exactly that?